modules/standard/Socket.md

```diff
@@ -0,0 +1,210 @@
+  ```python
+  proc tcpListener.accept() : tcpConn
+  ```
+- `close`,
+  calling close on `tcpListener` will close the socket server and it won't accept anymore new connections.
+  ```python
+  proc tcpListener.close()
+  ```
+- `family` returns the family of socket `AF_INET` or `AF_INET6`
+- `addr` will return an `ipAddr` instance providing details about the address and port at which socket is listening.
+  ```python
+  proc tcpListener.addr() : ipAddr
+  ```
+
+The additional support with `tcpConn` and `tcpListener` will be to enable or disable two of the prominent optimization algorithm for TCP Protocol that are:
+
+- `nagle`
+  This method will set/unset TCP_NODELAY for provided socket.
+  ```python
+  proc nagle(socket:tcpConn, tcpNodeDelay:bool = 1)
+  proc nagle(socket:tcpListener, tcpNodeDelay:bool = 1);
+  proc nagle(socket:udpSocket, tcpNodeDelay:bool = 1);
+  ```
+- `quickack`
+  This method will set/unset TCP_QUICKACK for provided socket.
+  ```python
+  proc quickack(socket:tcpConn, tcpQuickack:bool = 1)
+  proc qucikack(socket:tcpListener, tcpQuickack:bool = 1);
+  proc quickack(socket:udpSocket, tcpQuickack:bool = 1);
+  ```
+
+## UDP Support
+
+Support for UDP protocol will be through a separate constructor for `udpSocket`. They need not be binded or connected as they are connectionless socket.
+```python
+const socketServ:udpSocket = new udpSocket(family=IPFamily.IPv4)
+```
+
+A generic `bind` function will be used which will take in any of the socket three object and bind it to provided `host` , `port`.
+
+**Method**
+
+```
+proc bind(socketServ, addr:ipAddr, reuseAddr=true)
+```
+
+**Parameters**
+
+```
+- socket : tcpListener|tcpConn|udpSocket - socket object to be bound
+- addr: contains address info about where to bind the socket.
+- reuseaddr: Boolean - optional parameter, default value = true
+```
+
+Bind is responsible for binding any socket to a specified address and port. Here the socket can be anything either a UDP or a TCP Socket. listen under the hood will can in bind to bind `tcpListener` to address and port. With either tcpConn or udpSocket that isn't a requirement as the system assigns any free port and localhost address to them by default.
+
+While `tcpConn` is in a connected state when they are created `udpSocket` are connectionless they are just waiting for the user to use them for sending and receiving purpose only. But for sending and receiving purpose user might require a socket to have a known address that is when bind will be useful for `udpSocket`.
+
+## Data transfer on UDP
+
+UDP Sockets will have method to call `recvFrom` on it this will return the `addr`, `port` of client socket and `data` being written to the socket.
+
+**Method:**
+```python
+proc socketServ.recvFrom() : (addr:ipAddr, data:Generic)
```

Comment:
Chatting offline with Michael about the interface, he suggested (and I agree) that it might worth emphasizing why UDP differs from TCP's approach that's channel based.

Reply:
sure I will add in details about why the change was made to use raw functions.